containers/message-parser/app/main.py

```
@@ -238,14 +239,11 @@ async def upload_schema(
         }
 
     # Convert Pydantic models to dicts so they can be serialized to JSON.
-    for field in input.parsing_schema:
-        field_dict = input.parsing_schema[field].dict()
-        if "secondary_schema" in field_dict and field_dict["secondary_schema"] is None:
-            del field_dict["secondary_schema"]
-        input.parsing_schema[field] = field_dict
```

I have no memory at all of working on this part of the code 🤔 . Upon further investigation that is because my work on refactoring did not in fact touch any of the endpoints for querying or putting configs. I think I agree with your assumption that this code is intended to remove empty or unnecessary secondary field portions, but the only way I can see they would even get there in the first place is if the user submitted a config and just didn't understand they didn't need to do that. I almost wonder if instead of cleaning after the fact, we have some kind of validation at the input level that rejects any schemas uploaded with blank secondary schema dicts? @DanielPaseltiner curious to hear your thoughts on that.

---

So they do populate...somewhere in the PUT or in the GET that I couldn't directly find, because the schemas I was uploading were inserting the NULLs. 

I tend to agree though that if someone uploads a schema with NULL'd secondary_schema, we should probably throw an error or warning.

---

I think Pydantic actually ends up introducing these empty dicts that cause problems. So we are really cleaning up the mess that one of our tools creates, not necessarily one that the client sends us. @robertandremitchell discovered this when attempting upload schemas that don't define a `secondary_schema`.

---

ahhh okay, that would explain it. Then yeah, the new utils function should cover this well. thank you!